src/Uno.UI.RemoteControl/HotReload/ClientHotReloadProcessor.PartialReload.cs

```diff
@@ -0,0 +1,169 @@
+﻿#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+using Uno.Extensions;
+using Uno.Foundation.Logging;
+using Uno.UI.RemoteControl.HotReload.Messages;
+using System.Runtime.Loader;
+using System.Runtime.CompilerServices;
+using System.Diagnostics;
+
+namespace Uno.UI.RemoteControl.HotReload
+{
+	partial class ClientHotReloadProcessor
+	{
+		private Dictionary<string, Type>? _mappedTypes;
+
+		private bool _supportsLightweightHotReload;
+
+		private Task? _updatingTypes;
+		private readonly object _updatingTypesGate = new();
+
+		private void InitializePartialReload()
+		{
+			_supportsLightweightHotReload = (_msbuildProperties?.TryGetValue("TargetFramework", out var targetFramework) ?? false)
+				&& (_msbuildProperties?.TryGetValue("BuildingInsideVisualStudio", out var buildingInsideVisualStudio) ?? false)
+				&& buildingInsideVisualStudio.Equals("true", StringComparison.OrdinalIgnoreCase)
+				&& (
+					targetFramework.Contains("-android")
+					|| targetFramework.Contains("-ios"));
```

Comment:
Do we need to support catalyst here too?

Reply:
Indeed we should!

Reply:
Actually, no, because catalyst is not supported VS windows, and on mac we go through vscode which does not use this code path.